code/mission/import/xwingmissionparse.cpp

```diff
@@ -623,6 +623,231 @@ void parse_xwi_flightgroup(mission *pm, const XWingMission *xwim, const XWMFligh
+	this determines how wide the minefield is.Minefields are always a square formation,
+	centered over their starting point.For example,
+	a value of 3 will create a 3x3 grid of mines.**/
+
+		
+
+	// object position and orientation
+	// NOTE: Y and Z are swapped after all operartions are perfomed
+
+	float objectPosX = oj->object_x;
+	float objectPosY = oj->object_y;
+	float objectPosZ = oj->object_z;
+
+	int mine_dist = 400; // change this to change the distance between the mines
+	/** The minefield is a square 2d along two planes (a,b) centred on the given position **/
+	float offsetAxisA = 0 -(mine_dist / 2 * (number_of_objects - 1)); // (- the distance to centre the grid)
+	float offsetAxisB = 0 -(mine_dist / 2 * (number_of_objects - 1));
+		
+	matrix orient;
+	xwi_determine_object_orient(&orient, oj);
+
+	// Copy objects in Parse_objects to set for name checking below
+	// This only needs to be done fully once per object group then can be added to after each new object
+	SCP_set<SCP_string> objectNameSet;
+	for (int n = 0; n < Parse_objects.size(); n++) {
+		objectNameSet.insert(Parse_objects[n].name);
+	}
+
+	// now begin to configure each object in the group (mines multiple)
+
+	for (int a = 0; a < number_of_objects; a++) { // make an a-b 2d grid from the mines
+		offsetAxisA += (mine_dist * a); // add a new row to the grid
+		for (int b = 0; b < number_of_objects; b++) { // for each increment along the a plane, add mines along b plane
+			offsetAxisB += (mine_dist * b);  // for each new row populate the column
+			
+			/** Now convert the grid (a,b) to the relavenat formation ie. (x,y) or (z,y) etc **/
+			auto ojxyz = xwi_determine_mine_formation_position(oj, objectPosX, objectPosY, objectPosZ, offsetAxisA, offsetAxisB);
+			vm_vec_scale(&ojxyz, 1000); // units are in kilometers (after processing by xwinglib which handles the
+										// factor of 160), so scale them up
+
+			p_object pobj;
+			SCP_totitle(pobj.name);
+			pobj.orient = orient;
+			pobj.pos = ojxyz;
+			
+			int team = 0; //default civilian/team none
+			if (number_of_objects > 1)
+				team = 2; //mines are always hostile
+
+			int ship_class = ship_info_lookup(object_type);
+			if (ship_class < 0) {
+				Warning(LOCATION, "Unable to determine ship class for Object Group with type %s", object_type);
+				ship_class = 0;
+			}
+			auto sip = &Ship_info[ship_class];
+
+			// regular name, regular suffix
+			char base_name[NAME_LENGTH];
+			char suffix[NAME_LENGTH];
+			strcpy_s(base_name, object_type);
+
+			end_string_at_first_hash_symbol(base_name);
+
+			int n = 1;
+			char suffix[NAME_LENGTH];
+			sprintf(suffix, NOX(" %d"), n);
+
+			// Keep searching through the names until a unique one is discovered
+			while (!unique_object_suffix(objectNameSet, base_name, suffix))
+			{
+				n++;
+				sprintf(suffix, NOX(" %d"), n);
+				if (n > objectNameSet.size()) { // if Size = all the elements + null then a name should be found
+					goto noNameError;
+				}
+			}
```

Comment:
Most/all of this should be in the name picker function.  Call it `xwi_determine_space_object_name` rather than `unique_object_suffix`